core/systems/graphite/leaderboard_settings_form.ex

```diff
@@ -0,0 +1,92 @@
+defmodule Systems.Graphite.LeaderboardSettingsForm do
+  use CoreWeb.LiveForm, :fabric
+  use Fabric.LiveComponent
+
+  import Frameworks.Pixel.Form
+
+  alias Frameworks.Pixel.Text
+
+  alias Systems.{
+    Graphite
+  }
+
+  @visibility_options [
+    %{id: "public", value: "public", active: true},
+    %{id: "private", value: "private", active: false},
+    %{id: "private with date", value: "private with date", active: false}
+  ]
+
+  @allow_anonymous_options [
+    %{id: "false", value: false, active: true},
+    %{id: "true", value: true, active: false}
+  ]
+
+  @impl true
+  def update(%{id: id, leaderboard: leaderboard}, socket) do
+    changeset = Graphite.LeaderboardModel.changeset(leaderboard, %{})
+
+    {
+      :ok,
+      socket
+      |> assign(
+        id: id,
+        leaderboard: leaderboard,
+        changeset: changeset,
+        visibility_options: @visibility_options,
+        allow_anonymous_options: @allow_anonymous_options
+      )
+    }
+  end
+
+  @impl true
+  def handle_event("save", %{"leaderboard_model" => attrs}, socket) do
+    %{assigns: %{leaderboard: leaderboard}} = socket
+
+    attrs =
+      if attrs["metrics"] do
+        Map.put(attrs, "metrics", string_to_metrics(attrs["metrics"]))
+      else
+        attrs
+      end
+
+    {
+      :noreply,
+      socket
+      |> save(leaderboard, attrs)
+    }
+  end
+
+  defp save(socket, entity, attrs) do
+    changeset = Graphite.LeaderboardModel.changeset(entity, attrs)
+
+    socket
+    |> save(changeset)
+  end
+
+  defp metrics_to_string(nil), do: ""
```

Comment:
Could also be done using a virtual Ecto field. For example see `Systems.Org.NodeModel` `type` and `type_string` fields. Not a big deal though.